org.eclipse.scanning.device.ui/src/org/eclipse/scanning/device/ui/device/DetectorView.java

```diff
@@ -332,12 +342,58 @@ protected void configure() {
 		try {
 			IRunnableDevice<Object> device = dservice.getRunnableDevice(info.getName());
 			Object model = info.getModel();
+
+			// Pass null to 'clear' the validation results view
+			selectionProvider.fireSelection(new StructuredSelection(new ValidateResults(info.getName(), null)));
+			
+			Object validateReturn = device.validateWithReturn(model);
+
+			ValidateResults validateResults = new ValidateResults(info.getName(), validateReturn);
+				
+			try {
+				PlatformUI.getWorkbench().getDisplay().asyncExec(new Runnable() {
+					@Override
+					public void run() {
+						try {
+							if (PageUtil.getPage().findView(ValidateResultsView.ID) == null) {
+								IViewPart viewPart = PageUtil.getPage().showView(ValidateResultsView.ID);
+								if (viewPart instanceof ValidateResultsView) {
+									ValidateResultsView validateResultsView = (ValidateResultsView)viewPart;
+									validateResultsView.update(validateResults);
+								}
+							} else {
+								PageUtil.getPage().showView(ValidateResultsView.ID);
+							}
+						} catch (PartInitException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)